부산대 BE_오선재 Step0 - 기본 코드 준비

5seonjae opened the pull request:
이전에 진행했던 상품 옵션 코드들을 옮겨왔습니다.

이후에 진행할 미션 내용들 말고도 이전 미션에 제가 추가하지 못했던 수정사항들은 이후 PR 에 반영하도록 하겠습니다.

잘 부탁드립니다!

5seonjae committed:
feat: 이전 미션 구현 코드 새 프로젝트로 이동

Gyuchool commented on .gitignore:
```diff
@@ -38,3 +38,6 @@ out/
 .DS_Store
+
+### 환경 변수 파일 ###
+.env
```
해당 파일이 없어서 제가 실패하는 것 같습니다..🥲 따로 빼신 이유가 있으실까요?
<img width="1122" height="350" alt="스크린샷 2025-07-23 오후 2 57 30" src="https://github.com/user-attachments/assets/8eabc85c-4d34-465e-ad78-6d8122764dcb" />

5seonjae replied:
.env 파일을 따로 뺀 이유는 민감 정보 ( jwt 시크릿 키, 만료시간 ) 를 깃 리포지토리에 커밋하지 않기 위해서입니다.

```.env
jwt.secret=ZQ06uDt8FTkJrY3G2u/qAc4boHirPmQPLmRiAetJgwk=
jwt.expiration-ms=3600000
```

.env 파일입니다. 이 파일이 있으면 정상적으로 실행이 될겁니다.
코드 리뷰에 있어서 .env 파일을 git 에 포함시키지 않았던 점이 불편하실 수 있으셨을 것 같습니다. 죄송합니다...

.env 파일을 커밋해도 된다면 코드 리뷰에 차질이 발생하지 않게 .env 파일도 git 에 추가하도록 하겠습니다.

5seonjae replied:
.env 파일을 커밋해도 된다면 코드 리뷰에 차질이 발생하지 않게 .env 파일도 git 에 추가하도록 하겠습니다.

Gyuchool replied:
Readme같은 곳에 적용방법을 적어두시면 다른 개발자들이 로컬에서 쉽게 실행할 수 있을 것 같습니다!👍

5seonjae replied:
넵 다음 PR 때 README.md 에 적용방법을 적어두도록 하겠습니다!

Gyuchool reviewed (changes requested):
안녕하세요 선재님!
이번 미션을 함께하게 될 김규철이라고 합니다🙇‍♂️ 잘부탁드립니다😊

코드가 실행이 안되어서 확인 부탁드립니다! (env파일이 제가 없어서 그런지 실패하네요..!)

5seonjae committed:
fix(import.sql): import 스크립트 필드명 오타 수정

- import.sql에서 잘못된 컬럼명(option_name, option_quantity)을 실제 테이블 컬럼명(name, quantity)으로 변경하여 데이터 삽입 오류 해결
- DB 스키마(import.sql)와 테이블 구조 일관성 확보

5seonjae committed:
fix(validation): imageUrl 검증 순서 고정 및 테스트 데이터 수정

- ProductCreateRequestDto, ProductUpdateRequestDto에 @GroupSequence({DTO.class, NotBlankCheck.class, PatternCheck.class}) 추가해 imageUrl 검증 순서를 NotBlank → Pattern으로 고정
- NotBlankCheck, PatternCheck 마커 인터페이스 생성
- updateProduct_fail_invalidImageUrl 테스트에서 price 값을 -1000 → 1000으로 수정해 imageUrl 검증이 실행되도록 보장

5seonjae commented:
추가로 수정할 부분이 있어서 수정했습니다!

Gyuchool reviewed (approved):
안녕하세요 선재님
테스트 및 빌드 정상 동작하는 거 확인해서 머지하도록 하겠습니다~!

다음 단계에서 뵐게요!
